src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -137,7 +146,8 @@ private EncryptionKey addSiteKey(int siteId, Duration activatesIn) throws Except
         // force refresh manually
         this.keyProvider.loadContent();
 
-        return addSiteKeys(Arrays.asList(siteId), activatesIn, siteKeyExpiresAfter).get(0);
+        // Set cutoffTime to be 0 here as we don't want to filter out the keys that have passed the cutoff time
+        return addSiteKeys(Arrays.asList(siteId), activatesIn, siteKeyExpiresAfter, Duration.ofDays(0)).get(0);
```

Comment:
maybe my understanding is wrong - i take it you mean you wanna KEEP ALL the keys regardless the cutoff time config or is this wrong? if you set the cutoff time to 0, as soon as the key expires it will be filtered out in the code below right?

```
 final List<EncryptionKey> keys = this.keyProvider.getSnapshot().getActiveKeySet().stream()
                .sorted(Comparator.comparingInt(EncryptionKey::getId))
                .filter(k -> now.compareTo(k.getExpires().plus(cutoffTime.toDays(), ChronoUnit.DAYS)) < 0)
                .collect(Collectors.toList());
```

Reply:
You are absolutely right... I will make the change. 

I'm surprised we don't have tests to cover this 🤔